tests/notificationprofile/test_views.py

```diff
@@ -74,26 +82,220 @@ def test_notification_profile_can_update_timeslot_without_changing_pk(self):
+                "type": "object",
+                "required": ["phone_number"],
+                "properties": {
+                    "phone_number": {
+                        "type": "string",
+                        "title": "Phone number",
+                        "description": "The phone number is validated and the country code needs to be given.",
+                    }
+                },
+                "$id": "http://testserver/json-schema/sms",
+            }
+        }
+
+        response = self.user1_rest_client.get(path=f"/api/v2/notificationprofiles/media/sms/json_schema/")
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+        self.assertEqual(response.data, schema)
+
+    def test_can_get_all_media(self):
+        response = self.user1_rest_client.get(path=f"/api/v2/notificationprofiles/media/")
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+        self.assertEqual(len(response.data), 2)
+        self.assertEqual(set([medium["slug"] for medium in response.data]), set(["sms", "email"]))
+
+    def test_can_get_medium(self):
+        response = self.user1_rest_client.get(path=f"/api/v2/notificationprofiles/media/email/")
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+        self.assertEqual(response.data["name"], "Email")
+        response = self.user1_rest_client.get(path=f"/api/v2/notificationprofiles/media/sms/")
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+        self.assertEqual(response.data["name"], "SMS")
+
+
+@tag("API", "integration")
+class DestinationViewTests(APITestCase):
+    ENDPOINT = "/api/v2/notificationprofiles/destinations/"
+
+    def setUp(self):
+        disconnect_signals()
+        user1 = PersonUserFactory()
+
+        self.user1_rest_client = APIClient()
+        self.user1_rest_client.force_authenticate(user=user1)
+
+        # Default email destination is automatically created with user
+        self.synced_email_destination = user1.destinations.get()
+        self.non_synced_email_destination = DestinationConfigFactory(
+            user=user1,
+            media=Media.objects.get(slug="email"),
+            settings={"email_address": "test@example.com", "synced": False},
+        )
+        self.sms_destination = DestinationConfigFactory(
+            user=user1,
+            media=Media.objects.get(slug="sms"),
+            settings={"phone_number": "+4747474747"},
+        )
+
+        # Hook up destinations to user
+        timeslot1 = TimeslotFactory(user=user1, name="Never")
+        self.notification_profile1 = NotificationProfileFactory(user=user1, timeslot=timeslot1)
+
+        self.notification_profile1.destinations.set([self.synced_email_destination])
+        self.notification_profile1.destinations.set([self.non_synced_email_destination])
+        self.notification_profile1.destinations.set([self.sms_destination])
```

Comment:
What is going on here?
```suggestion
        self.notification_profile1.destinations.set([self.synced_email_destination, self.non_synced_email_destination, self.sms_destination])
```

Reply:
I didn't write the tests, I just split 'em up and changed setUp.